sdk/src/androidTest/java/com/mapbox/maps/OfflineTest.kt

```diff
@@ -0,0 +1,627 @@
+          .geometry(TOKYO)
+          .descriptors(
+            listOf(
+              offlineManager.createTilesetDescriptor(
+                TilesetDescriptorOptions.Builder()
+                  // load style pack here and not separately to reduce amount of code
+                  .styleURI(STYLE)
+                  .stylePackOptions(
+                    StylePackLoadOptions.Builder()
+                      .glyphsRasterizationMode(GlyphsRasterizationMode.IDEOGRAPHS_RASTERIZED_LOCALLY)
+                      .metadata(Value(STYLE_PACK_METADATA))
+                      .build()
+                  )
+                  // set min zoom twice less in order to make sure tile in loaded map view
+                  // will be fully visible and not somewhere on tile edge
+                  .minZoom((ZOOM.toInt() / 2).toByte())
+                  .maxZoom(ZOOM.toInt().toByte())
+                  .build()
+              )
+            )
+          )
+          .metadata(Value(TILE_REGION_METADATA))
+          .tileLoadOptions(
+            TileLoadOptions.Builder()
+              .networkRestriction(NetworkRestriction.NONE)
+              .criticalPriority(false)
+              .acceptExpired(true)
+              .build()
+          )
+          .build(),
+        {}
+      ) {
+        if (it.isValue) {
+          it.value?.let { latch.countDown() }
+        } else {
+          Logger.e(TAG, "Error when loading tile: ${it.error?.toString()}")
+        }
+      }
+    }
+    if (!latch.await(30, TimeUnit.SECONDS)) {
+      throw TimeoutException()
+    }
+  }
+
+  private fun prepareMapView(observer: Observer, eventList: List<String>) {
+    var mapboxMap: MapboxMap
+    rule.scenario.onActivity {
+      it.runOnUiThread {
+        val context = InstrumentationRegistry.getInstrumentation().targetContext
+        // uses default resource options under the hood
+        val mapView = MapView(
+          context = context,
+          mapInitOptions = MapInitOptions(
+            context = context,
+            initialCameraOptions = CameraOptions.Builder()
+              .center(TOKYO)
+              .zoom(ZOOM)
+              .build()
+          )
+        )
+        mapboxMap = mapView.getMapboxMap()
+        mapboxMap.subscribe(observer, eventList)
+        mapboxMap.loadStyleUri(STYLE)
+        it.setContentView(mapView)
+        mapView.onStart()
+      }
+    }
+  }
+
+  private fun switchAirplaneMode() {
+    InstrumentationRegistry.getInstrumentation().uiAutomation.executeShellCommand("am start -a android.settings.AIRPLANE_MODE_SETTINGS")
+    val device = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+    device.waitForIdle()
+    // adopted for current CI device
+    device.findObject(UiSelector().textContains("Airplane")).click()
+    device.waitForIdle()
+    val latch = CountDownLatch(1)
+    // need to wait some time for pretty slow CI device
+    latch.await(5, TimeUnit.SECONDS)
+    device.pressBack()
+  }
```

Comment:
can we move this utility function to some base class so it can be reused? We could also use the function provided by common.

Reply:
What is the function provided by common?

Reply:
Sorry I just noticed you have a separate test for it, `TileStoreMapboxSwitch`.